docs/resources/deployment.md

```diff
@@ -0,0 +1,854 @@
+---
+page_title: "Elastic Cloud: ec_deployment Resource"
+description: |-
+  Provides an Elastic Cloud deployment resource, which allows deployments to be created, updated, and deleted.
+---
+
+# Resource: ec_deployment
+
+Provides an Elastic Cloud deployment resource, which allows deployments to be created, updated, and deleted.
+
+~> **Note on Elastic Stack versions** Using a version prior to `6.6.0` is not supported.
+
+-> **Note on regions and deployment templates** Before you start, you might want to read about [Elastic Cloud deployments](https://www.elastic.co/guide/en/cloud/current/ec-create-deployment.html) and check the [full list](https://www.elastic.co/guide/en/cloud/current/ec-regions-templates-instances.html) of regions and deployment templates available in Elasticsearch Service (ESS).
+
+-> **Note on Elasticsearch topology IDs** Since the addition of data tiers, each Elasticsearch topology block requires the `"id"` field to be set. The accepted values are set in the deployment template that you have chosen, but values are closely related to the Elasticsearch data tiers. [Learn more abut Elasticsearch data tiers](https://www.elastic.co/guide/en/elasticsearch/reference/current/data-tiers.html). For a complete list of all the supported values, refer to the deployment template definition used by your deployment.
```

Comment:
There is no more topology block in Elasticsearch. Shall we reformulate this or maybe remove it completely?